Source/SwiftLintFramework/Rules/Metrics/ViewControllerFileLengthRule.swift

```diff
@@ -0,0 +1,52 @@
+import SourceKittenFramework
+
+public struct ViewControllerFileLengthRule: ConfigurationProviderRule {
+    public var configuration = FileLengthRuleConfiguration(warning: 400, error: 1000)
+
+    private let viewControllerPattern = "ViewController"
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "viewcontroller_file_length",
+        name: "Viewcontroller File Length",
+        description: "ViewControllers should not span too many lines.",
+        kind: .metrics,
+        nonTriggeringExamples: [
+            Example(repeatElement("print(\"swiftlint\")\n", count: 399).joined())
+        ],
+        triggeringExamples: [
+            Example(repeatElement("print(\"swiftlint\")\n", count: 601).joined()),
+            Example((repeatElement("print(\"swiftlint\")\n", count: 600) + ["//\n"]).joined()),
+            Example(repeatElement("print(\"swiftlint\")\n\n", count: 1001).joined())
+        ].skipWrappingInCommentTests()
+    )
+
+    public func validate(file: SwiftLintFile) -> [StyleViolation] {
+        let isViewController = file.structureDictionary.isViewController()
+
+        guard isViewController else { return [] }
+
+        var lineCount = file.lines.count
+        let hasViolation = configuration.severityConfiguration.params.contains {
+            $0.value < lineCount
+        }
+
+        if hasViolation && configuration.ignoreCommentOnlyLines {
+            lineCount = file.lineCountWithoutComments()
+        }
+
+        for parameter in configuration.severityConfiguration.params where lineCount > parameter.value {
```

Comment:
If I'm not mistaking, this finds only the first param where line count is greater than parameter.value. Is that expected?
If so, you can simplify that by:
```swift
let firstMatch = configuration
    .severityConfiguration
    .params
    .first(where: { $0.value > lineCount)

...
```